### ISSUE TITLE
does sshtunnel allow local port forwarded to remote host?

### ISSUE THREAD

**Issue by diego-vasarelli** [pahaz/sshtunnel#111]
Hi,

I would like to use sshtunnel for a fully python implementation of these shell commands: 
1. `ssh –L 9999:127.0.0.1:8888 user@server` so I can use server:8888 service on my local:9999
2. `ssh –R 7777:127.0.0.1:6666 user@server` so server can use service in local:6666 internally, as server:7777 

The aim to use sshtunnel is to have this feature both on windows and linux without adding extra packet.
The first point is correctly satisfied by this:

```python
from sshtunnel import SSHTunnelForwarder

###   ssh –L 9999:127.0.0.1:8888 user@server

server = SSHTunnelForwarder(
    '10.0.0.1', #server host
    ssh_username="user",
    ssh_password="pass",
    remote_bind_address=('127.0.0.1', 8888),
    local_bind_address=('127.0.0.1', 9999)
)

server.start()

print(server.local_bind_port)  # show assigned local port
# work with `SECRET SERVICE` through `server.local_bind_port`.

server.stop()
```

I would like to ask if there is a simple way to obtain also the second point.
I've tried to change configuration parameters but with no success. I've also tried to use paramiko library but seems to not work on windows as sshtunnel does.

Thanks in advance 
Hi,

I would like to use sshtunnel for a fully python implementation of these shell commands: 
1. `ssh –L 9999:127.0.0.1:8888 user@server` so I can use server:8888 service on my local:9999
2. `ssh –R 7777:127.0.0.1:6666 user@server` so server can use service in local:6666 internally, as server:7777 

The aim to use sshtunnel is to have this feature both on windows and linux without adding extra packet.
The first point is correctly satisfied by this:

```python
from sshtunnel import SSHTunnelForwarder

###   ssh –L 9999:127.0.0.1:8888 user@server

server = SSHTunnelForwarder(
    '10.0.0.1', #server host
    ssh_username="user",
    ssh_password="pass",
    remote_bind_address=('127.0.0.1', 8888),
    local_bind_address=('127.0.0.1', 9999)
)

server.start()

print(server.local_bind_port)  # show assigned local port, forced to 9999

server.stop()
```

I would like to ask if there is a simple way to obtain also the second point.
I've tried to change configuration parameters but with no success. I've also tried to use paramiko library but seems to not work on windows as sshtunnel does.

Thanks in advance

**Comment by thorstenkampe** [comment]
The question is actually about Remote Port Forwarding. Remote Port Forwarding is described in the Readme (second example).

**Comment by fernandezcuesta** [comment]
Hi, Remote port forwarding is not implemented, but it's a nice to have feature for `sshtunnel`.
Anyway, have a look at [paramiko remote forward example](https://github.com/paramiko/paramiko/blob/master/demos/rforward.py) for details.

**Comment by diego-vasarelli** [comment]
@thorstenkampe I saw the second example but it seems to not cover exactly my problem. It seems to be about a third server who provide a service on the ssh link between client and first server. Can you provide an example?

@fernandezcuesta  I came to sshtunnel from paramiko because example forward.py example not worked for me; so I decided to not use it. I'll give a try to Asyncssh.

Thanks for the help.

**Comment by adamwojt** [comment]
Yeah `-R` is super confusing. I spent hours to figure out how to simply do what `-R` accomplishes with plain `ssh`
